### ISSUE TITLE
add `Eq` instance

### ISSUE THREAD

**Issue by m4lvin** [jwaldmann/haskell-obdd#12]
It would be nice to have an easy equivalence (which on BDDs is the identity) check.
Concretely, I would like to have `instance Eq a => instance (Eq (OBDD a))` or so.

**Comment by jwaldmann** [comment]
Yes, semantically this makes sense, and it'd be nice to have in the API, since BDDs _are_ canonical forms.

Note that there is `OBDD.Operation.equiv` already.

Any implementation of `(==)` still would need to traverse both graphs: BDDs representing the same function are  isomorphic, but their node numbers might differ.

Until there's a better implementation, we can add
```
instance Ord v => Eq (OBDD v) where
  f == g = null $ not $ equiv f g
```

In "real" BDD libraries, this question does not occur: they use a BDD base, and then semantic equality can be detected directly by a pointer comparison.
